app/models/application_record.rb

```diff
@@ -13,4 +13,12 @@ class ApplicationRecord < ActiveRecord::Base
   # FIXME: UI code - decorator support
   extend MiqDecorator::Klass
   include MiqDecorator::Instance
+
+  # API Support
+  virtual_column :slug, :type => :string
+
+  def slug
```

Comment:
@Fryguy Should this logic be moved into a separate file and included here like the `Ar*` modules above.

Is `slug` a descriptive enough method name.  Is `url_slug` or `href_slug` more useful?  Just a thought.